File: GildedRose.cs
using System.Collections.Generic;
using GildedRoseRefactoringKata.QualityUpdater;

namespace GildedRoseRefactoringKata
{
    public class GildedRose
    {
        private readonly IList<Item> _items;

        public GildedRose(IList<Item> items)
        {
            _items = items;
        }

        public void UpdateQuality()
        {
            foreach (var item in _items)
            {
                var qualityUpdater = QualityUpdaterFactory.Build(item);
                qualityUpdater.Update();
            }
        }
    }
}

File: src/gilded_rose.js
import { Item } from "./item";

class Shop {
  constructor(items = []) {
    this.items = items;
  }
  updateQuality() {
    this.items = this.items.map(item => item.updateQuality());
    return this.items;
  }
}

module.exports = {
  Item,
  Shop
}
